### ISSUE TITLE
[FEATURE] Allow users to run a validation metric on a model post-training, optionally on a different validation dataset

### ISSUE THREAD

**Issue by tmostak** [h2oai/h2o-llmstudio#237]
### 🚀 Feature

Allow users to run any supported validation metric, against any valid dataframe, after training.

### Motivation

Sometimes after training a model it is useful to be able to test the model on either a different validation set, or using a different validation metric than used when training. As an example motivating use case of the latter, I've found that computing the BLEU metric is significantly more expensive than the perplexity metric, to the extent that when training a model with validation points every 0.1 epochs, the model training can take up to 3X longer than if using Perplexity as the validation metric. Hence, it would be nice to use Perplexity as the validation metric during training, but after training, to be able to test the model using the BLEU metric. Same goes for wanting to try a different validation set post-training.

**Comment by psinger** [comment]
Thanks for the issue.

This is actually already possible, but the workflow is a bit hidden, here is how it works:

1. After you finished an experiment, you click on New Experiment from the kebab menu on the right hand side:

![image](https://github.com/h2oai/h2o-llmstudio/assets/1677826/77b9a543-ae65-4e4f-b7a1-50a3309125c9)

2. Then, you click on the following setting on top:

![image](https://github.com/h2oai/h2o-llmstudio/assets/1677826/6ea7023d-6c1f-4831-a1db-23f4218e67ef)

3. This will now load the previous weights, and now you can change eval dataset, metric, and whatever you want. And to only do evaluation, without any retraining, you can set the epochs to 0:

![image](https://github.com/h2oai/h2o-llmstudio/assets/1677826/3cc02529-3e5a-41dd-91c6-e719abb4bfc7)

Please let us know if this works for your use-case which absolutely makes sense. Actually it is a great idea to focus on perplexity for training as generate is incredibly slow.

**Comment by pascal-pfeiffer** [comment]
@sherenem , let's add this to the FAQ probably

**Comment by psinger** [comment]
> Not a big deal either way, but does this approach result in a full copy of the model weights to the new model?

yeah it does, it will create a new experiment practically

> I still think that a dedicated way to kick off a new validation metric for an existing model (perhaps from the kebab menu) would be nice as it would be more discoverable and be clearly associated with an existing model, but this is nit-picking, the above workaround unblocks me.

makes sense, will add it to a list of suggestions, I think maybe not always re-saving the weights and linking to an old experiment could be a solution

> Btw, really nice work building such a awesome and useful platform, it's by far the easiest and most intuitive to use of anything I've tried so far, while still giving lots of powerful customization options for more advanced use cases.

great you like it! please keep posting suggestions / issues along the way

**Comment by psinger** [comment]
Added the feedback here: https://github.com/h2oai/h2o-llmstudio/issues/238
thx!
